src/plugins/userImportExport/src/java/org/jivesoftware/openfire/plugin/OpenfireExporter.java

```diff
@@ -236,7 +236,7 @@ public boolean validate(InputStream file) {
         try {
           userName = Stringprep.nodeprep(userName);
 
-          if (isUserProviderReadOnly) {
+          if (!isUserProviderReadOnly && (password != null)) {
```

Comment:
Isn't the `password != null` check already done on line 235?

Reply:
You are right, but then is the question, which can be removed? Password is needed to create a user. but not for adding Roster entries. If you compare this to the Xep227Exporter, it would be logical to remove the first check.

Reply:
if the user does not get created, it would seem to not made sense to add roster items to a user that does not exist?  I am unsure though

Reply:
This is handled. When the user is not known, an exception is thrown by the usermanager.getUser(), which is handled later on. With a readOnly UserProvider it would be used to update the roster information, where the Users are already are migrated in the LDAP for example.